tensordict/tensordict.py

```diff
@@ -1626,9 +1627,26 @@ def flatten_keys(
         self, separator: str = ".", inplace: bool = False
     ) -> TensorDictBase:
         to_flatten = []
+        already_flat = []
         for key, meta_value in self.items_meta():
             if meta_value.is_tensordict():
                 to_flatten.append(key)
+            else:
+                already_flat.append(key)
+
+        for key in already_flat:
+            if separator in key:
+                _existing_val = None
+                try:
```

Comment:
try/except are expensive if the except is reached (they're very cheap otherwise).
https://stackoverflow.com/a/2522013/4858862
Since we just 'pass' in the except it might occur that this operation is repeated very frequently in the code, thereby affecting performance.